client/packages/system/src/Encounter/DetailView/Toolbar.tsx

```diff
@@ -267,11 +267,18 @@ export const Toolbar: FC<ToolbarProps> = ({
                     value={DateUtils.getDateOrNull(endDatetime ?? null)}
                     onChange={date => {
                       const endDatetime = date
-                        ? DateUtils.formatRFC3339(date)
+                        ? DateUtils.formatRFC3339(
+                            new Date(
+                              new Date(startDatetime ?? '').setHours(
+                                date.getHours(),
+                                date.getMinutes()
+                              )
+                            )
+                          )
                         : undefined;
                       if (endDatetime) {
                         setEndDatetime(endDatetime);
-                        onChange({ endDatetime });
+                        onChange({ startDatetime, endDatetime });
```

Comment:
Is this startDatetime needed, seems to work without?

Reply:
Yeah not need since it's getting the correct date now